realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -521,6 +521,99 @@ private void checkEmpty(String str) {
     }
 
     /**
+     * Returns the column indices for the given field name. If a linked field is defined, the column index for
+     * each field is returned
+     *
+     * @param fieldDescription fieldName or link path to a field name.
+     * @param validColumnTypes Legal field type for the last field a
+     * @return list of column indices.
+     */
+    // TODO: consider another caching strategy so linked classes are included in the cache.
+    long[] getColumnIndices(String fieldDescription, RealmFieldType... validColumnTypes) {
+        if (fieldDescription == null || fieldDescription.equals("")) {
+            throw new IllegalArgumentException("Non-empty fieldname must be provided");
+        }
+        if (fieldDescription.endsWith(".")) {
+            throw new IllegalArgumentException("Illegal field name. It cannot end with a '.'");
+        }
```

Comment:
It work work anyway. I added this because we had a unit test catching it, but you are right, we should also check for a starting `.`